core/src/main/scala/pureconfig/backend/ConfigFactoryWrapper.scala

```diff
@@ -22,15 +22,17 @@ object ConfigFactoryWrapper {
     unsafeToEither(ConfigFactory.load())
 
   /** @see `com.typesafe.config.ConfigFactory.parseFile()` */
-  def parseFile(path: Path): Either[ConfigReaderFailures, Config] =
-    unsafeToEither(ConfigFactory.parseFile(path.toFile))
+  def parseFile(path: Path): Either[ConfigReaderFailures, Config] = {
+    if (!path.toFile.isFile || !path.toFile.canRead) Left(ConfigReaderFailures(CannotReadFile(path)))
```

Comment:
This check catches many common cases, but it does not ensure that the file can indeed be read. The only sure way to know that is reading the file ourselves and pass its contents to `ConfigFactory.parseString`, but then we would need to care with other things, like dealing with charsets. Do you think this check is enough, at least for now?